hashicorp/dev-portal
# fix: no mobile scroll when menu is open

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksfix-mobile-no-scroll-hashicorp.vercel.app/vault/docs) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202227071718207) 🎟️

## 🗒️ What

This PR fixes an issue where if one scrolled down the page and opened the sidebar, it may not render. Also if the sidebar was open and visible, it could be scrolled past. 

## 🛠️ How

I created a hook `useNoScrollBody` that accepts a trigger param, when this trigger is true, the body won't be scrollable. I also adjusted the positioning of the mobile sidebar a bit. 

## 🧪 Testing

- [Visit a page ](https://dev-portal-git-ksfix-mobile-no-scroll-hashicorp.vercel.app/vault/docs) in simulator or adjust your browser window to <tablet view. 
- Open the sidebar and scroll down. You should only be able to scroll to the bottom of the sidebar contents. 
- Close the sidebar and scroll a ways down the page, open it again to see that it renders in the correct position. 

- Also the search results component was updated to use this shared hook. Please [visit a docs page](https://dev-portal-git-ksfix-mobile-no-scroll-hashicorp.vercel.app/vault/docs) with search on a mobile view, type something in so the results panel shows up - there should be no regressions and the body shouldn't be scrollable.
